### PR TITLE
Switch from ConcurrentHashMap to Collections.synchronizedMap() in GenerateTypeAdapter

### PR DESCRIPTION
**Pull request by hzsweers**
ConcurrentHashMap bugs on some versions of android: https://code.google.com/p/android/issues/detail?id=170073

### CODE REVIEW

**Comment by rharter** [comment]
Got some test failures.

**Comment by hzsweers** [comment]
This is what I get for editing the file directly on GitHub :)

File: auto-value-gson-annotations/src/main/java/com/ryanharter/auto/value/gson/GenerateTypeAdapter.java [hunk 1]
```diff
@@ -27,7 +28,7 @@
 
   TypeAdapterFactory FACTORY = new TypeAdapterFactory() {
     private final Map<Class<?>, Constructor<? extends TypeAdapter>> adapters =
-        new ConcurrentHashMap<>();
+        Collections.synchronizedMap(new HashMap<Class<?>, Constructor<? extends TypeAdapter>>());
```
Comment (NightlyNexus) [inline auto-value-gson-annotations/src/main/java/com/ryanharter/auto/value/gson/GenerateTypeAdapter.java]: btw, [LinkedHashMap is always better than HashMap.](https://publicobject.com/2016/02/08/linkedhashmap-is-always-better-than-hashmap/)

Comment (hzsweers) [reply auto-value-gson-annotations/src/main/java/com/ryanharter/auto/value/gson/GenerateTypeAdapter.java]: Ordering doesn't matter here nor are we testing it, I'm not too worried about it :)

**Comment by hzsweers** [comment]
We have an error-prone check against it internally, but yeah seems a good candidate

File: auto-value-gson-annotations/src/main/java/com/ryanharter/auto/value/gson/GenerateTypeAdapter.java (hunk 1 above)
Comment (JakeWharton) [reply auto-value-gson-annotations/src/main/java/com/ryanharter/auto/value/gson/GenerateTypeAdapter.java]: Ordering _always_ matters regardless on whether you explicitly rely on it or not. That's the point of the post.

Comment (hzsweers) [reply auto-value-gson-annotations/src/main/java/com/ryanharter/auto/value/gson/GenerateTypeAdapter.java]: I don't feel strongly about it in this case, but done in 24fc3b7

**Comment by hzsweers** [comment]
reran travis a couple times, seems it's been flaky the past few days. Green now

**Pull request merged**